src/client/app/shared/canvas/canvas.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, HostListener } from '@angular/core';
+import { CardComponent } from '../../shared/index';
+import { CanvasService, Point } from './canvas.service';
+
+/**
+ * This class represents the Canvas component.
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'uxg-canvas',
+  templateUrl: 'canvas.component.html',
+  styleUrls: ['canvas.component.css'],
+  directives: [CardComponent],
+  providers: [CanvasService]
+})
+export class CanvasComponent {
+  // Whether or not panning is in progress.
+  panning: boolean = false;
+  // The last point seen during the pan that is currently in progress.
+  lastPanPnt: Point = null; //TODO(eyuelt): make this nullable after TS2 update
+
+  constructor(public canvasService: CanvasService) {
+  }
+
+  // TODO(eyuelt): should this use HostListener or the template event binding?
+  onMousewheel(event: WheelEvent) {
+    const zoomScale = 1 + (event.deltaY * -0.01);
+    const zoomPnt = {
```

Comment:
I would inline this object literal. For me, the inline version is more readable.